docs/enhancements/arm64-cpu-architecture.md

```diff
@@ -0,0 +1,152 @@
+            "rhcos_image": "https://mirror.openshift.com/pub/openshift-v4/aarch64/dependencies/rhcos/pre-release/latest-4.9/rhcos-live.aarch64.iso",
+            "rhcos_rootfs": "https://mirror.openshift.com/pub/openshift-v4/aarch64/dependencies/rhcos/pre-release/latest-4.9/rhcos-live-rootfs.aarch64.img",
+            "rhcos_version": "49.84.202106272247-0"
+      }
+   }
+}
+```
+
+For backwards compatibility, the current format of openshift-version should still be supported.
+So older versions (<4.9), that don't support arm64 CPU architecture, could use the old format x86_64 architecture only.
+
+### Cluster creation and ISO generation
+
+When creating a new cluster, `cpu_architecture` property should be supported to define which hosts can be registered to it. This property would be added to cluster-create-params and stored in cluster upon creation. If not specified, defaulted to x86_64 as a fallback.
+
+An example for the swagger schema:
+
+```
+cpu_architecture:
+   type: string
+   default: 'x86_64'
+   description: The CPU architecture of the image (x86_64/arm64/etc)
+```
+
+When generating a discovery ISO, the proper RHCOS/Release images should be used according to CPU architecture specified in the cluster. The `cpu_architecture` property should be stored in `infraenv` after the ISO is generated.
+
+### CPU_ARCHITECTURES environment variable
+
+The supported CPU architectures will be defined in CPU_ARCHITECTURES environment variable. The default should be: 'x86_64,arm64'.
+As arm64 is not supported for older OCP versions (<4.9), it could be validated against OPENSHIFT_VERSIONS only for OCP >=4.9.
+
+### User Stories
+
+#### Story 1
+
+As an Assisted-Installer user, I need to install a cluster on non x86_64 hosts.
+
+#### Story 2
+
+As an Assisted-Installer user, I need to install a Single-Node OpenShift cluster on an arm64 host.
+
+### Implementation Details/Notes/Constraints [optional]
+
+* The current focus is for SNO flows, but other scenarios could be supported.
+
+* Late binding is not covered by this enhancement, but decoupling the discovery ISO generation from the cluster should be simple enough. I.e. adding `cpu_architecture` property to `infraenv_create_params` and override the architecture defined in the cluster. Also, architecture validation should be probably added to host discovery conditions (i.e. to ensure that the host CPU architecture is similar to the cluster's architecture).
+
+### Open Questions
+
+#### kube-api considerations
+
+##### ClusterImageSet
+Since we extract the openshift-baremetal-install binary in the assisted-service (to create the manifests of the cluster), the x86_64 OCP release image has to be used. Which means that both arm64 and x86_64 release images must be provided.
+  * Should we perhaps add another URL for the x86_64 release image in ClusterImageSet?
+  * Maybe we could assume that a x86_64 release image is available by default for each 'y-stream' version?
```

Comment:
What do you mean by "for each `y-stream` version"? I'm not sure how that relates to a ClusterImageSet, which references a specific openshift release.